### ISSUE TITLE
Unmatched bracket breaks line wrapping

### ISSUE THREAD

**Issue by evmar** [jrblevin/markdown-mode#148]
```
;; Version: 2.1
;; Package-Version: 20160629.459
```

Input:

```
foo bar
foo bar
foo bar
foo bar
foo bar
foo bar
[
foo bar
foo bar
foo bar

foo bar
foo bar
foo bar
foo bar
```

Set a fill column:
`M-x set-variable RET fill-column RET 72 RET`

Then hit `M-q` while the cursor is in the first block.  You'll see that the `[` causes odd wrapping, and also that the subsequent block cannot be wrapped.

Using backticks around the `[` doesn't help either.  I couldn't find any workaround.  This came up in a place where I was discussing ANSI escapes, which are text sequences that look like `\e[34m`.

**Comment by martine** [comment]
Possible workaround: use `&#x5b;` in place of the bracket.

**Comment by martine** [comment]
`&#x5b;` in a code block doesn't become the `[`.

**Comment by jrblevin** [comment]
Thanks for reporting this. I've confirmed it. It would seem to be related to some of the variables that control filling behavior, to avoid breaking lines in the middle of an inline link, but I'll have to look into it further.

**Comment by jrblevin** [comment]
I can't reproduce this anymore.  In fact, I think this was fixed a while ago by 3f11a4a, which was in response to #161.  The problem was that `markdown-mode` mistakenly thought the point was inside an link, due to the unclosed square bracket, but I fixed that.  Please reopen this issue if this didn't fix the problem.
